lib/opal/cli_runners/server.rb

```diff
@@ -30,8 +37,8 @@ def exit_status
         nil
       end
 
-      def app(source)
-        lambda do |env|
+      def build_app(source)
+        app = lambda do |env|
```

Comment:
Probably it makes sense to extract this lambda to a separate callable object:
``` ruby
class App
  def initialize(source)
    @source = source
  end

  BODY = <<-HTML.freeze
    <!doctype html>
    <html>
      <head>
        <meta charset="utf-8"/>
        <script src="/cli_runner.js"></script>
      </head>
    </html>
  HTML

  def call(env)
    case env['PATH_INFO']
    when '/'
      [200, {}, [BODY]]
    when '/cli_runner.js'
      [200, {'Content-Type' => 'text/javascript'}, [source]]
    else
      [404, {}, [body]]
    end
  end
end

def build_app(source)
  app = App.new(source)

  if static_folder
    app = Rack::Cascade.new(
      Rack::Static.new(:urls => [''], root: static_folder),
      app,
    )
  end

  app
end
```

Reply:
The server stuff slipped in by error, anyway I'm following your advice. 
Someday I'd like to rewrite the cli runners and make them pluggable from outside…